src/lib.rs

```diff
@@ -445,6 +452,33 @@ pub fn watchdog_enabled(unset_env: bool, usec: &mut u64) -> bool {
     }
 }
 
+fn monotonic_time_usecs() -> u128 {
+    use libc::{timespec, CLOCK_MONOTONIC, clock_gettime};
+
+    let mut monotonic_time = timespec {
+        tv_sec: 0,
+        tv_nsec: 0,
+    };
+
+    let result = unsafe {
+        clock_gettime(CLOCK_MONOTONIC, &mut monotonic_time)
+    };
+
+    if result != 0 {
+        // Should we return this as Result or Option?
+        return 0;
+    }
+
+    // First, nanosecond / 1000 -> microseconds.
+    let lower_microseconds = (monotonic_time.tv_nsec.unsigned_abs() / 1000) as u128;
```

Comment:
Do we need `unsigned_abs`? I think we can return `i64`. If the OS gives us a negative value, that's on the OS.

Reply:
We have to combine the two i64's into a u128. So we can't represent negatives. If the os gives us a negative, we have a bad time ahead ....

Reply:
We can still return `i128` (or probably even `i64`, that's what `nix` does), so no problem with the sign.

Reply:
If we get negative nsec and pos sec, what do we do? Both negative? It gets messy pretty darn fast ... That's why I think it's not wise to muck with signs here and we strip it out of the problem.

Reply:
We add them normally and don't worry about it? If someone says "2 s and -100 ms", that's weird, but it's 1.9, not 2.1 s. (I know we have ns here, but it's the same thing).

Reply:
We would need to do i64 -> i128 then i128 to u128. So I still think it gets a bit messy. I will do it if you want. 

I don't think it would break the monotonic nature - if anything using the negatives if they were there might break it since that could create a step back when the result should have been positive. (Aren't unix api's a glorious hot mess? 🤣 )

Reply:
> then i128 to u128

Why? We can return `i128` or `i64`.

> if anything using the negatives if they were there might break it

Well, if the OS gives us -1000 s now, then 998 s two seconds later, who are we to turn that into 1000 and 998?

Reply:
I used u128 to match Rust's much more sensible, https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_micros function.

Reply:
Oh, but that's a `Duration`, which can't really be negative, depending on how you define it. Our value is more like `Instant`, which has no constraints like this besides actually being monotonic. Maybe it starts from the UNIX epoch, maybe it doesn't. Maybe it's negative and the system shuts down when it reaches 0. Of course, that's unlikely, but being "more correct" with less code is good. [Bonus reading](https://devblogs.microsoft.com/oldnewthing/20180228-00/?p=98125).

Reply:
Ill go with the i128 then.